src/SearchEngine.java

```diff
@@ -0,0 +1,111 @@
+import strategies.SearchStrategy;
+import strategies.StrategyALL;
+import strategies.StrategyANY;
+import strategies.StrategyNONE;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class SearchEngine {
+
+    Map<String, ArrayList<Integer>> map = new HashMap<>();
+    ArrayList<String> list = new ArrayList<>();
+    static Scanner scanner = new Scanner(System.in);
+
+    public void run(File file) {
+        ArrayList<String> peopleList = readFile(file);
+        Map<String, ArrayList<Integer>> words = mapWords(peopleList);
```

Comment:
Why do you need the `words` map when you have `map` one?

Reply:
fixed